### PR TITLE
DPL-939- Pacbio runcreate store change from VueX to Pinia

### PR DESCRIPTION
**Pull request by seenanair**
This includes following changes

 -  Pinia store creation for Pacbio runcreate - `usePacbioRunCreateStore` (/stores/pacbioRunCreate)
 -  Pinia store creation for root store - `useRootStore` (/stores/index)
 -  Unit test cases added for 
      - usePacbioRunCreateStore
      -  useRootStore
 -  Refactoring of all components using pacbioRunCreate store
      - PacbioRunWell
      - PacbioRunPlateItem
      - PacbioRunPlateList
      - PacbioRunShow
      - PacbioRunWellEdit
      - PacbioRunWellDefaultEdit
      - PacbioRunIndex
      - PacbioPoolList
      - PacbioPoolSelectedList
      - PacbioPoolTubeItem
      - PacbioRunInfoEdit
  - Unit test case update for all components using pacbioRunCreate store

### CODE REVIEW

File: tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js [hunk 1]
```diff
@@ -37,56 +38,83 @@ const smrtLinkVersions = {
...
+ * 'options' allows to define initial state of store while instantiating the component.
+ *
+ * @param {*} options - options to be passed to the createTestingPinia method for creating a mock instance of pinia
+ * options type is
+ * {state :{},stubActions: boolean, plugins:[]}
+ *
+ * @param {*} props - props to be passed to the component while mounting
+ *
+ */
+function mountWithStore(options, props) {
```
Comment (stevieing) [inline tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js:53]: same as above using mountWithStore

File: tests/unit/components/pacbio/PacbioRunWellDefaultEdit.spec.js [hunk 2]
```diff
@@ -32,23 +26,52 @@ const smrtLinkVersions = {
...
+/**
+ * Helper method for mounting a component with a mock instance of pinia, with the given 'options'.
+ * 'options' allows to define initial state of store while instantiating the component.
+ *
+ * @param {*} options - options to be passed to the createTestingPinia method for creating a mock instance of pinia
+ * options type is
+ * {state :{},stubActions: boolean, plugins:[]}
+ *
+ */
+function mountWithStore(options) {
```
Comment (stevieing) [inline tests/unit/components/pacbio/PacbioRunWellDefaultEdit.spec.js:38]: same as above using mountWithStore

File: tests/unit/components/pacbio/PacbioRunWellEdit.spec.js [hunk 3]
```diff
@@ -25,7 +26,37 @@ const props = {
...
+/**
+ * Helper method for mounting a component with a mock instance of pinia, with the given 'options'.
+ * 'options' allows to define initial state of store while instantiating the component.
+ *
+ * @param {*} options - options to be passed to the createTestingPinia method for creating a mock instance of pinia
+ * options type is
+ * {state :{},stubActions: boolean, plugins:[]}
+ *
+ */
+function mountWithStore(options) {
```
Comment (stevieing) [inline tests/unit/components/pacbio/PacbioRunWellEdit.spec.js:38]: same as above using mountWithStore

**Review by stevieing — CHANGES_REQUESTED** [review]
Pretty good. Just a couple of things that could be tidied up.

**Review by stevieing — COMMENTED** [review]
I have also noticed when I run this locally that the default SMRT Link Versions is not selected. It is there when I run it from develop.

File: tests/unit/components/labware/PacbioRunPlateList.spec.js [hunk 4]
```diff
@@ -1,13 +1,44 @@
...
+/**
+ * Helper method for mounting a component with a mock instance of pinia, with the given 'options'.
+ * 'options' allows to define initial state of store while instantiating the component.
+ *
+ * @param {*} options - options to be passed to the createTestingPinia method for creating a mock instance of pinia
+ * options type is
+ * {state :{},stubActions: boolean, plugins:[]}
+ *
+ */
+function mountWithStore(options) {
```
[the remark this answers is not in the document]
Comment (seenanair) [reply tests/unit/components/labware/PacbioRunPlateList.spec.js:21]: i thought about it, but then I am not sure whether mount supports stubbing dynamic components passed on it.

[the remark this answers is not in the document]
Comment (seenanair) [reply tests/unit/components/labware/PacbioRunPlateList.spec.js]: I had a look into this. We can have a  generic function as follows in tesrHelper. But to optimise the code, we might still require a global function in each components that calls this, with all the custom parameters that are common between test cases. So, I think it doesn't bring much of an advantage. Please share your views.

`function mountWithStore(
  component,
  usePiniaStoreFn,
  storeNames,
  { initialState = {}, stubActions = false, plugins = [] } = {},
  props = {},
  stubs = {},
) {
  const wrapper = mount(component, {
    global: {
      plugins: [
        createTestingPinia({
          initialState,
          stubActions,
          plugins,
        }),
      ],
      stubs,
    },
    router,
    props,
  })
  const store = usePiniaStoreFn()
  return { wrapper, store }
}`

**Comment by seenanair** [comment]
> I have also noticed when I run this locally that the default SMRT Link Versions is not selected. It is there when I run it from develop.

Just to make sure, I hope you mentioned the 'New run' page?

**Review by stevieing — APPROVED** [review]

**Pull request merged**